src/api/NestedRestriction.ts

```diff
@@ -19,7 +19,7 @@ export class NestedRestriction {
     }
 
     /** The clauses containing the nested restrictions and their logical operators. */
-    public clauses = [] as Clause[];
+    public clauses?: Clause[];
```

Comment:
Doesn't the assignments below effectively make this property non-optional? So the `?` doesn't really have any effect here? (I'm still learning typescript so could be wrong here)

Reply:
ah, yup... I fiddled with that one so much to make things happy I ended up going around it 3 times